fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -569,14 +690,33 @@ def upload(app_id = nil, dir = nil, package_path: nil, asset_path: nil)
       password_placeholder = @jwt.nil? ? 'YourPassword' : nil
       jwt_placeholder = @jwt.nil? ? nil : 'YourJWT'
 
-      command = @transporter_executor.build_upload_command(@user, @password, actual_dir, @provider_short_name, @jwt)
-      UI.verbose(@transporter_executor.build_upload_command(@user, password_placeholder, actual_dir, @provider_short_name, jwt_placeholder))
+      # Handle AppStore Connect API
+      use_api_key = !@api_key.nil?
+      api_key_placeholder = nil unless use_api_key
+      api_key_placeholder = { key_id: "YourKeyID", issuer_id: "YourIssuerID" } if use_api_key
```

Comment:
Code-style nitpick: merging those two lines into a single one using a ternary condition will make this easier to read/reason about imho:

```suggestion
      api_key_placeholder = use_api_key ? { key_id: "YourKeyID", issuer_id: "YourIssuerID" } : nil
```